ui/components/component-library/modal-footer/modal-footer.tsx

```diff
@@ -55,6 +55,7 @@ export const ModalFooter: ModalFooterComponent = React.forwardRef(
         >
           {onCancel && (
             <Button
+              data-testid="mm-modal-cancel"
```

Comment:
Instead of adding a `data-testid` directly to the reusable component, it's better to pass it through the props and apply it where the component is being used. This approach ensures that each `data-testid` is unique to the feature or context where the component is implemented. It's important because it helps prevent flaky tests by avoiding conflicts and ensuring more precise targeting of elements during testing. Unique test IDs tied to specific use cases improve test reliability and maintainability.
```
<ModalFooter cancelButtonProps={{'data-testid': 'feature-modal-cancel-button'}} ... />
```

Reply:
+1 on what George said here